src/main/java/org/betonquest/betonquest/quest/condition/height/HeightConditionFactory.java

```diff
@@ -0,0 +1,71 @@
+package org.betonquest.betonquest.quest.condition.height;
+
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.logger.BetonQuestLogger;
+import org.betonquest.betonquest.api.logger.BetonQuestLoggerFactory;
+import org.betonquest.betonquest.api.quest.condition.PlayerCondition;
+import org.betonquest.betonquest.api.quest.condition.PlayerConditionFactory;
+import org.betonquest.betonquest.api.quest.condition.online.OnlineConditionAdapter;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.instruction.variable.Variable;
+import org.betonquest.betonquest.quest.PrimaryServerThreadData;
+import org.betonquest.betonquest.quest.condition.PrimaryServerThreadPlayerCondition;
+import org.betonquest.betonquest.quest.registry.processor.VariableProcessor;
+
+/**
+ * Factory for {@link HeightCondition}s.
+ */
+public class HeightConditionFactory implements PlayerConditionFactory {
+
+    /**
+     * Logger factory to create a logger for conditions.
+     */
+    private final BetonQuestLoggerFactory loggerFactory;
+
+    /**
+     * Data used for condition check on the primary server thread.
+     */
+    private final PrimaryServerThreadData data;
+
+    /**
+     * The variable processor.
+     */
+    private final VariableProcessor variableProcessor;
+
+    /**
+     * Create the height factory.
+     *
+     * @param loggerFactory     the logger factory
+     * @param data              the data used for checking the condition on the main thread
+     * @param variableProcessor the variable processor
+     */
+    public HeightConditionFactory(final BetonQuestLoggerFactory loggerFactory, final PrimaryServerThreadData data, final VariableProcessor variableProcessor) {
+        this.loggerFactory = loggerFactory;
+        this.data = data;
+        this.variableProcessor = variableProcessor;
+    }
+
+    @Override
+    public PlayerCondition parsePlayer(final Instruction instruction) throws InstructionParseException {
+        final BetonQuestLogger log = loggerFactory.create(HeightCondition.class);
+        final String string = instruction.next();
+        final Variable<Number> height = new Variable<>(variableProcessor, instruction.getPackage(), string, (value) -> {
+            try {
+                final double parsedValue;
+                if (value.matches("-?\\d+\\.?\\d*")) {
+                    parsedValue = Double.parseDouble(value);
+                } else {
+                    parsedValue = instruction.getLocation(value).getValue(null).getY();
```

Comment:
I think it would be better to implement it like this
```suggestion
                    parsedValue = VariableLocation.parse(value).getY();
```
of cause we need to make parse public then but i dont see an issue with that. The same can may then be with the Vector and World